mm2src/coins/hd_wallet/storage/mod.rs

```diff
@@ -219,25 +220,27 @@ impl Default for HDWalletCoinStorage {
 }
 
 impl HDWalletCoinStorage {
+    // TODO: Since hd_wallet_rmd160 is unique for a device, do we use it as db_id too? or we can just use mm2 shared_db_id and use hd_wallet_rmd160 for primary key as it's currently done
```

Comment:
Can we combine hd wallet with external wallets? I don't think it is possible, is it? @shamardy

Reply:
> For instance, if the same seed is used in both Keplr and mm2, and a swap is initiated using Keplr, then the app was closed midway through the swap. Upon restarting, if the Tendermint coin is activated using mm2 seed and the address enabled for swaps is the same address that was used before in Keplr, the swap will complete successfully.

I guess this is one of the test cases that should be tested by QA/GUI team one this PR is ready for testing
c.c. @smk762 @naezith